### ISSUE TITLE
I have a question about open vocabulary detection [BUG] <title>

### ISSUE THREAD

**Issue by romanticegg** [QwenLM/Qwen-VL#58]
### 是否已有关于该错误的issue或讨论？ | Is there an existing issue / discussion for this?

- [X] 我已经搜索过已有的issues和讨论 | I have searched the existing issues / discussions

### 该问题是否在FAQ中有解答？ | Is there an existing answer for this in FAQ?

- [X] 我已经搜索过FAQ | I have searched FAQ

### 当前行为 | Current Behavior

I want to test the open vocabulary detection task on QWen-VL model, but I find that I can't let it output all detection boxes at the specific catogary through the instruct 'all shoes' or 'all clothes'. 

### 期望行为 | Expected Behavior

How can it output all detection boxes at the specific catogary?

### 运行环境 | Environment

```Markdown
- OS:Ubuntu
- Python:3.9
- Transformers:4.31.0
- PyTorch:1.12.0
- CUDA (`python -c 'import torch; print(torch.version.cuda)'`):11.4
```

**Comment by jinze1994** [comment]
Hi, @romanticegg, as more supplements:

In the second training stage(the multi-task training) of Qwen-VL, we apply the grounding tasks on GRIT, refcocoX datasets, where there is only a very small amount of multiple boxes data. As a result, Qwen-VL has almost no ability to output multiple bounding boxes.

We added parts of [COCO-Instance](https://cocodataset.org/#download) data in the SFT, so Qwen-VL-Chat learns to output all bounding boxes when it comes to the keyword `all` in prompts.

**Comment by romanticegg** [comment]
> Hi, @romanticegg~ Thanks for your interests in our work. You can try the Qwen-VL-Chat model.

thanks, it works

**Comment by romanticegg** [comment]
> Hi, @romanticegg, as more supplements:
> 
> In the second training stage(the multi-task training) of Qwen-VL, we apply the grounding tasks on GRIT, refcocoX datasets, where there is only a very small amount of multiple boxes data. As a result, Qwen-VL has almost no ability to output multiple bounding boxes.
> 
> We added parts of [COCO-Instance](https://cocodataset.org/#download) data in the SFT, so Qwen-VL-Chat learns to output all bounding boxes when it comes to the keyword `all` in prompts.

By the way, I want to know how to get the confidence of the detection boxes? Thanks a lot

**Comment by romanticegg** [comment]
> > > Hi, @romanticegg, as more supplements:
> > > In the second training stage(the multi-task training) of Qwen-VL, we apply the grounding tasks on GRIT, refcocoX datasets, where there is only a very small amount of multiple boxes data. As a result, Qwen-VL has almost no ability to output multiple bounding boxes.
> > > We added parts of [COCO-Instance](https://cocodataset.org/#download) data in the SFT, so Qwen-VL-Chat learns to output all bounding boxes when it comes to the keyword `all` in prompts.
> > 
> > 
> > By the way, I want to know how to get the confidence of the detection boxes? Thanks a lot
> 
> You can calculate the loss on tokens of detection boxes. A similar code can be referenced [HERE](https://github.com/QwenLM/Qwen-VL/blob/master/eval_mm/seed_bench/eval.py#L153).

emmm, I want to know when there is no ground truth, how can I filter the boxes, is there any threshold in config to set?

**Comment by hhaAndroid** [comment]
> Hi, @romanticegg, as more supplements:
> 
> In the second training stage(the multi-task training) of Qwen-VL, we apply the grounding tasks on GRIT, refcocoX datasets, where there is only a very small amount of multiple boxes data. As a result, Qwen-VL has almost no ability to output multiple bounding boxes.
> 
> We added parts of [COCO-Instance](https://cocodataset.org/#download) data in the SFT, so Qwen-VL-Chat learns to output all bounding boxes when it comes to the keyword `all` in prompts.

For the detection task, a single image will output over 100 bounding boxes and categories. Is there any problem with the sequence output of VLLM in this case? Do you have any verification?

**Comment by dydxdt** [comment]
> > > > Hi, @romanticegg, as more supplements:
> > > > In the second training stage(the multi-task training) of Qwen-VL, we apply the grounding tasks on GRIT, refcocoX datasets, where there is only a very small amount of multiple boxes data. As a result, Qwen-VL has almost no ability to output multiple bounding boxes.
> > > > We added parts of [COCO-Instance](https://cocodataset.org/#download) data in the SFT, so Qwen-VL-Chat learns to output all bounding boxes when it comes to the keyword `all` in prompts.
> > > 
> > > 
> > > By the way, I want to know how to get the confidence of the detection boxes? Thanks a lot
> > 
> > 
> > You can calculate the loss on tokens of detection boxes. A similar code can be referenced [HERE](https://github.com/QwenLM/Qwen-VL/blob/master/eval_mm/seed_bench/eval.py#L153).
> 
> emmm, I want to know when there is no ground truth, how can I filter the boxes, is there any threshold in config to set?
@romanticegg @jinze1994 I also wonder where I can set the threshold in the config for the detection task. I debug the test scripts and I find '<ref>xx</ref><box>xx</box>' is just output after 'tokenizer.decode'. I'm new to the transformer code. Can you explain it? Thank you!

**Comment by dydxdt** [comment]
> > > > Hi, @romanticegg, as more supplements:
> > > > In the second training stage(the multi-task training) of Qwen-VL, we apply the grounding tasks on GRIT, refcocoX datasets, where there is only a very small amount of multiple boxes data. As a result, Qwen-VL has almost no ability to output multiple bounding boxes.
> > > > We added parts of [COCO-Instance](https://cocodataset.org/#download) data in the SFT, so Qwen-VL-Chat learns to output all bounding boxes when it comes to the keyword `all` in prompts.
> > > 
> > > 
> > > By the way, I want to know how to get the confidence of the detection boxes? Thanks a lot
> > 
> > 
> > You can calculate the loss on tokens of detection boxes. A similar code can be referenced [HERE](https://github.com/QwenLM/Qwen-VL/blob/master/eval_mm/seed_bench/eval.py#L153).
> 
> emmm, I want to know when there is no ground truth, how can I filter the boxes, is there any threshold in config to set?

@romanticegg @jinze1994 I also wonder where I can set the threshold in the config for the detection task. I debug the test scripts and I find 'xxxx' is just output after 'tokenizer.decode'. I'm new to the transformer code. Can you explain it? Thank you!

**Comment by dydxdt** [comment]
> @romanticegg @jinze1994 I also wonder where I can set the threshold in the config for the detection task. I debug the test scripts and I find 'xxxx' is just output after 'tokenizer.decode'. I'm new to the transformer code. Can you explain it? Thank you!

@romanticegg @jinze1994 I also wonder where I can set the threshold in the config for the detection task. I debug the test scripts and I find '<ref>xxxx</ref><box>xxx</box>' is just output after 'tokenizer.decode'. I'm new to the transformer code. Can you explain it? Thank you!

**Comment by dydxdt** [comment]
> > > > > Hi, @romanticegg, as more supplements:
> > > > > In the second training stage(the multi-task training) of Qwen-VL, we apply the grounding tasks on GRIT, refcocoX datasets, where there is only a very small amount of multiple boxes data. As a result, Qwen-VL has almost no ability to output multiple bounding boxes.
> > > > > We added parts of [COCO-Instance](https://cocodataset.org/#download) data in the SFT, so Qwen-VL-Chat learns to output all bounding boxes when it comes to the keyword `all` in prompts.
> > > > 
> > > > 
> > > > By the way, I want to know how to get the confidence of the detection boxes? Thanks a lot
> > > 
> > > 
> > > You can calculate the loss on tokens of detection boxes. A similar code can be referenced [HERE](https://github.com/QwenLM/Qwen-VL/blob/master/eval_mm/seed_bench/eval.py#L153).
> > 
> > 
> > emmm, I want to know when there is no ground truth, how can I filter the boxes, is there any threshold in config to set?
> 
> @romanticegg @jinze1994 I also wonder where I can set the threshold in the config for the detection task. I debug the test scripts and I find 'xxxx' is just output after 'tokenizer.decode'. I'm new to the transformer code. Can you explain it? Thank you!

'xxx' means <ref>..</ref><box>...</box>

**Comment by dydxdt** [comment]
> > > > > Hi, @romanticegg, as more supplements:
> > > > > In the second training stage(the multi-task training) of Qwen-VL, we apply the grounding tasks on GRIT, refcocoX datasets, where there is only a very small amount of multiple boxes data. As a result, Qwen-VL has almost no ability to output multiple bounding boxes.
> > > > > We added parts of [COCO-Instance](https://cocodataset.org/#download) data in the SFT, so Qwen-VL-Chat learns to output all bounding boxes when it comes to the keyword `all` in prompts.
> > > > 
> > > > 
> > > > By the way, I want to know how to get the confidence of the detection boxes? Thanks a lot
> > > 
> > > 
> > > You can calculate the loss on tokens of detection boxes. A similar code can be referenced [HERE](https://github.com/QwenLM/Qwen-VL/blob/master/eval_mm/seed_bench/eval.py#L153).
> > 
> > 
> > emmm, I want to know when there is no ground truth, how can I filter the boxes, is there any threshold in config to set?
> 
> @romanticegg @jinze1994 I also wonder where I can set the threshold in the config for the detection task. I debug the test scripts and I find 'xxxx' is just output after 'tokenizer.decode'. I'm new to the transformer code. Can you explain it? Thank you!

'xxx' means < ref > .... < /ref > < box > .... < /box >
